plutus-core/plutus-core/src/PlutusCore/Flat.hs

```diff
@@ -85,6 +88,15 @@ implementations for them (if they have any constructors reserved for future use)
 By default, Flat does not use any space to serialise `()`.
 -}
 
+newtype AsSerialize a = AsSerialize
+    { unAsSerialize :: a
+    }
+
+instance Serialise a => Flat (AsSerialize a) where
+    encode = encode . serialise . unAsSerialize
+    decode = AsSerialize . deserialise <$> decode
+    size = error "Implement me"
```

Comment:
@michaelpj what about `size`?

Reply:
We're encoding a lazy bytestring, so presumably `size @ByteString`. Or you could call https://hackage.haskell.org/package/flat-0.4.4/docs/Flat-Encoder-Strict.html#v:sLazyBytes which is how `size` is implemented for lazy bytestrings.

Reply:
This:

```haskell
newtype AsSerialize a = AsSerialize
    { unAsSerialize :: a
    } deriving newtype (Serialise)

instance Serialise a => Flat (AsSerialize a) where
    encode = encode . serialise
    decode = deserialise <$> decode
    size = size . serialise
```

? Also pinging @kwxm and @raduom who I think also did something with `flat` (for context: we're trying to write a `Flat` instance for things that already have a `Serialize` instance).